pkg/kubectl/cmd/exec.go

```diff
@@ -177,6 +183,47 @@ func (p *ExecOptions) Validate() error {
 	return nil
 }
 
+func (o *StreamOptions) setupTTY() term.TTY {
+	t := term.TTY{
+		Parent: o.InterruptParent,
+		Out:    o.Out,
+	}
+
+	if !o.Stdin {
+		// need to nil out o.In to make sure we don't create a stream for stdin
+		o.In = nil
+		return t
+	}
+
+	t.In = o.In
+	if !o.TTY {
+		return t
+	}
+
+	if !t.IsTerminalIn() {
+		if o.Err != nil {
+			fmt.Fprintln(o.Err, "Unable to use a TTY - input is not a terminal or the right kind of file")
+		}
+
+		return t
+	}
+
+	// if we get to here, the user wants to attach stdin, wants a TTY, and o.In is a terminal, so we
+	// can safely set t.Raw to true
+	t.Raw = true
+
+	// use dockerterm.StdStreams() to get the right I/O handles on Windows
+	stdin, stdout, _ := dockerterm.StdStreams()
```

Comment:
This needs to be exposed so it is testable.  Would be sufficient to have a member function "func overrideStreams() (io.Reader, io.Writer, io.Writer)" that can be substituted.